### PR TITLE
Reset page number when sorting option `sortResetPage`

### LINKED ISSUE #6402: Reset page number to 1 while making a client side sort
### Bootstraptable version(s) affected

1.21.0

### Description

When making a sort, I would expect the table to reset to page number 1 automatically.
But this is not the case.

Steps to reproduce:
- https://live.bootstrap-table.com/code/marceloverdijk/12864
- Go to page 2
- Sot on any of the columns
- Actual result: Table stays on page 2
- Expected result: Reset page number to 1

The interesting part is that for **server side** sort the page number **is** reset.
This was implemented by @UtechtDustin in 2019 with this PR: https://github.com/wenzhixin/bootstrap-table/pull/4414

Client vs Server side sorting does not seem to have consistent behaviour.

### Example(s)

https://live.bootstrap-table.com/code/marceloverdijk/12864

### Possible Solutions

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by marceloverdijk**
**🤔Type of Request**
- [ ] **Bug fix**
- [ ] **New feature**
- [x] **Improvement**
- [ ] **Documentation**
- [ ] **Other**

**🔗Resolves an issue?**

Introduces a new `sortResetPage` option to reset the page number when sorting.
Note that current default behaviour and behaviour for server side sorting is kept as is.

Fixes #6402 

**📝Changelog**

- [x] **Core**
- [ ] **Extensions**

**💡Example(s)?**

https://live.bootstrap-table.com/code/marceloverdijk/13109

**☑️Self Check before Merge**

⚠️ Please check all items below before reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by marceloverdijk** [comment]
Also example added here: https://github.com/wenzhixin/bootstrap-table-examples/pull/469

File: site/docs/api/table-options.md [hunk 1]
```diff
@@ -1657,6 +1657,20 @@ The table options are defined in `jQuery.fn.bootstrapTable.defaults`.
...
 
+## sortResetPage
+
+- **Attribute:** `data-sort-reset-page`
+
+- **Type:** `Boolean`
+
+- **Detail:**
+
+  Set `true` to reset the page number when sorting
```
Comment (wenzhixin) [inline site/docs/api/table-options.md:1668]:
```suggestion
  Set `true` to reset the page number when sorting.
```

**Review by wenzhixin — APPROVED** [review]

**Comment by wenzhixin** [comment]
Thanks for your contribution!

**Comment by marceloverdijk** [comment]
@wenzhixin you are welcome. ps this one is related: https://github.com/wenzhixin/bootstrap-table/pull/6417

**Pull request merged**